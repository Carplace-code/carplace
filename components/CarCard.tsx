import Image from "next/image";

interface CarCardProps {
  title: string;
  price: string;
  platform: string;
  km: string;
  fuel: string;
  imageUrl?: string;
}

export default function CarCard({ title, price, platform, km, fuel, imageUrl }: CarCardProps) {
  return (
    <div className="rounded-lg border bg-white p-4 shadow-sm">
      <div className="mb-4 h-32 overflow-hidden rounded-md bg-gray-200">
        {imageUrl ? (
          <Image src={imageUrl} alt={title} className="h-full w-full object-cover" width={300} height={128} />
        ) : (
          <div className="flex h-full w-full items-center justify-center text-gray-400">No Image</div>
        )}
      </div>
      <h3 className="mb-1 font-semibold">{title}</h3>
      <p className="mb-2 text-sm text-gray-600">
        {price} • {platform}
      </p>
      <p className="text-sm text-gray-500">
        {km} • {fuel}
      </p>
      <button className="mt-4 w-full rounded-md bg-blue-600 py-2 text-sm text-white hover:bg-blue-700" type="button">
        View Details
      </button>
    </div>
  );
}
