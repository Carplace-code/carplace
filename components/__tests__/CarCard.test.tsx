import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import CarCard from "@/components/CarCard";

const mockCarProps = {
  id: "test-car-id",
  title: "Toyota Corolla 2024",
  price: "$18,990",
  platform: "Chileautos",
  km: "15,000 km",
  fuel: "Gas",
};

describe("CarCard", () => {
  test("renders CarCard with image", () => {
    const { container } = render(<CarCard {...mockCarProps} imageUrl="https://example.com/car-image.jpg" />);

    expect(container).toMatchSnapshot();
  });

  test("renders CarCard without image (shows fallback)", () => {
    const { container, getByText } = render(<CarCard {...mockCarProps} />);

    // Check that the "No Image" fallback is rendered
    expect(getByText("No Image")).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test("renders CarCard with all required props", () => {
    const { getByText } = render(<CarCard {...mockCarProps} />);

    expect(getByText("Toyota Corolla 2024")).toBeTruthy();
    expect(getByText("$18,990 • Chileautos")).toBeTruthy();
    expect(getByText("15,000 km • Gas")).toBeTruthy();
    expect(getByText("View Details")).toBeTruthy();
  });

  test("generates correct link href", () => {
    const { container } = render(<CarCard {...mockCarProps} />);

    const link = container.querySelector('a[href="/cars/test-car-id"]');
    expect(link).toBeTruthy();
  });
});
