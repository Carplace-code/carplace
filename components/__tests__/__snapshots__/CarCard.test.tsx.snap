// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CarCard > renders CarCard with image 1`] = `
<div>
  <div
    class="rounded-lg border bg-white p-4 shadow-sm"
  >
    <div
      class="mb-4 h-32 overflow-hidden rounded-md bg-gray-200"
    >
      <img
        alt="Toyota Corolla 2024"
        class="h-full w-full object-cover"
        data-nimg="1"
        decoding="async"
        height="128"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Fcar-image.jpg&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fcar-image.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fcar-image.jpg&w=640&q=75 2x"
        style="color: transparent;"
        width="300"
      />
    </div>
    <h3
      class="mb-1 font-semibold"
    >
      Toyota Corolla 2024
    </h3>
    <p
      class="mb-2 text-sm text-gray-600"
    >
      $18,990
       • 
      Chileautos
    </p>
    <p
      class="text-sm text-gray-500"
    >
      15,000 km
       • 
      Gas
    </p>
    <a
      class="mt-4 block w-full rounded-md bg-blue-600 py-2 text-center text-sm text-white hover:bg-blue-700"
      href="/cars/test-car-id"
    >
      View Details
    </a>
  </div>
</div>
`;

exports[`CarCard > renders CarCard without image (shows fallback) 1`] = `
<div>
  <div
    class="rounded-lg border bg-white p-4 shadow-sm"
  >
    <div
      class="mb-4 h-32 overflow-hidden rounded-md bg-gray-200"
    >
      <div
        class="flex h-full w-full items-center justify-center text-gray-400"
      >
        No Image
      </div>
    </div>
    <h3
      class="mb-1 font-semibold"
    >
      Toyota Corolla 2024
    </h3>
    <p
      class="mb-2 text-sm text-gray-600"
    >
      $18,990
       • 
      Chileautos
    </p>
    <p
      class="text-sm text-gray-500"
    >
      15,000 km
       • 
      Gas
    </p>
    <a
      class="mt-4 block w-full rounded-md bg-blue-600 py-2 text-center text-sm text-white hover:bg-blue-700"
      href="/cars/test-car-id"
    >
      View Details
    </a>
  </div>
</div>
`;
